virtual_ecosystem/models/soil/pools.py

```diff
@@ -482,9 +482,11 @@ def calculate_all_pool_updates(
         )
 
         # Calculate rate at which ammonium volatilises as ammonia
-        ammonia_volatilisation_rate = (
+        ammonia_volatilisation_rate = np.where(
```

Comment:
Why does this have to be conditional and not simply `self.constants.ammonia_volatilisation_rate_constant * self.pools.soil_n_pool_ammonium` . Is this to guarantee that it does not become negative?

Reply:
Yep because `self.pools.soil_n_pool_ammonium` can be negative, this step is needed to avoid negative volatilisation (which is nonsensical).